ddl/ddl_api.go

```diff
@@ -1625,7 +1625,9 @@ func (d *ddl) preSplitAndScatter(ctx sessionctx.Context, tbInfo *model.TableInfo
 	} else {
 		scatterRegion = variable.TiDBOptOn(val)
 	}
-	pi := tbInfo.GetPartitionInfo()
+	if pi == nil {
+		pi = tbInfo.GetPartitionInfo()
```

Comment:
This makes me confused.
What's the  difference between argument `pi` and `tbInfo.GetPartitionInfo()`, are't they the same thing?

Reply:
`pi` will be the added partition information when `AddTablePartitions ` call this function.

Reply:
Please add comment for this parameter, otherwise when somebody review this code later in the future, he will have the same question.

Reply:
If `AddTablePartitions ` calls this function,  `pi` isn't nil, so why we need to add the check of `if pi == nil`?

Reply:
`CreateTableWithInfo` will call this function too, then `pi` will be nil.

Reply:
Could we call `d.preSplitAndScatter(ctx, tbInfo, tbInfo.GetPartitionInfo())` to handle it in `CreateTableWithInfo`?